Improve Android and CD docs (#5694)
flutter/website: src/docs/deployment/cd.md

application is delivered to your beta testers and validated on a frequent basis
without resorting to manual workflows.

## CI/CD Options

There are a number of continuous integration (CI) and continuous delivery (CD)
options available to help automate the delivery of your application.

> ### All-in-one type of options, Flutter functionality built in XXX

Reviewer: ```suggestion
### All-in-one options with built-in Flutter functionality
```

* [Codemagic][]
* [Bitrise][]
* [Appcircle][]

> ### General CI options XXX
> * [Cirrus][]
> * [GitHub Actions][]
>     * Example: Flutter Gallery's [Github Actions workflows][]
>     * Example: [Github Action in Flutter Project][]
> * [Travis][]
> * [GitLab][]
> 
> This guide shows how to set up fastlane and then integrate it with 
> your existing testing and continuous integration (CI) workflows ("General CI options" above XXX).

Reviewer: ```suggestion
### Integrating fastlane with existing workflows

You can use fastlane with the following tooling:

* [GitHub Actions][]
    * Example: Flutter Gallery's [Github Actions workflows][]
    * Example: [Github Action in Flutter Project][]
* [Cirrus][]
* [Travis][]
* [GitLab][]

This guide shows how to set up fastlane and then integrate it with 
your existing testing and continuous integration (CI) workflows. 
For more information, see "Integrating fastlane with existing workflow".
```

application is delivered to your beta testers and validated on a frequent basis
without resorting to manual workflows.

## CI/CD Options

There are a number of continuous integration (CI) and continuous delivery (CD)
options available to help automate the delivery of your application.

> ### All-in-one type of options, Flutter functionality built in XXX

Author: Marked these headers with XXX for extra attention needed

* [Codemagic][]
* [Bitrise][]
* [Appcircle][]

### General CI options XXX
* [Cirrus][]
    * Example: Flutter framework repository's [Cirrus script][].
* [GitHub Actions][]
    * Example: Flutter Gallery's [Github Actions workflows][]
    * Example: [Github Action in Flutter Project][]
* [Travis][]
* [GitLab][]

This guide shows how to set up fastlane and then integrate it with 

> your existing testing and continuous integration (CI) workflows ("General CI options" above XXX).

Author: i.e. you likely wouldn't use fastlane unless you were using Cirrus, GitHub Actions, Travis, or Gitlab

3. Create the CI test script such as `.travis.yml` or `.cirrus.yml` in your
   repository root.
    * See [fastlane CI documentation][] for CI specific setup.
    * Shard your script to run on both Linux and macOS platforms.

>     * During the setup phase of the CI task:

Reviewer: ```suggestion
    * During the setup phase of the CI task, do the following:
```
